pkg/controller/deployment/sync.go

```diff
@@ -632,8 +592,12 @@ func calculateStatus(allRSs []*extensions.ReplicaSet, newRS *extensions.ReplicaS
 
 // isScalingEvent checks whether the provided deployment has been updated with a scaling event
 // by looking at the desired-replicas annotation in the active replica sets of the deployment.
-func (dc *DeploymentController) isScalingEvent(d *extensions.Deployment) (bool, error) {
-	newRS, oldRSs, err := dc.getAllReplicaSetsAndSyncRevision(d, false)
+//
+// rsList should come from getReplicaSetsForDeployment(d).
+// podMap should come from getPodMapForReplicaSets(rsList).
+// These are passed around to avoid repeating expensive API calls.
```

Comment:
The potentially expensive part is adoption and orphaning which happens in `getReplicaSetsForDeployment()`, and which I don't want to happen more than once per sync, from a semantic perspective.

The caching of the result from `getPodMapForReplicaSets()` is because it (for now at least; being discussed elsewhere in the PR) lists all pods regardless of selector. If you think this is premature optimization, I can remove `podMap` and just pass the `rsList` around.